### PR TITLE
Add crd-install helm hook to crds templates

### PR DESCRIPTION
**Pull request by smoya**
From Helm documentation:
> crd-install: Adds CRD resources before any other checks are run. This is used only on CRD definitions that are used by other manifests in the chart.

This PR aims to update agones helm charts crds adding the `crd-install` hook, ensuring then the installation of those to be done before other workloads.

This change also fixes third party charts with agones as dependency that deploy workloads related to those custom resources.

### CODE REVIEW

**Comment by markmandel** [comment]
You will want to run `make gen-install` to make sure the helm template lines up with the install.yaml

**Comment by smoya** [comment]
I am not sure about the reason of the test error. Adding annotations should not make the installation of helm charts to be broken. 
However, this annotation was release with helm v2.10.0.

**Comment by markmandel** [comment]
Ah! I was just looking at this - we are on helm [version 2.9.1](https://github.com/GoogleCloudPlatform/agones/blob/master/build/build-image/Dockerfile#L52).

I don't think there is any reason we can't upgrade - just need to do the work. Separate PR would be lovely :dancer:

**Comment by markmandel** [comment]
Helm upgrade complete. Let's try this again!

**Comment by markmandel** [comment]
@Kuqd any thoughts on below :cry: 

Did the consul lock not exit somehow?

I'm wondering if we got a new version of consul with the new cluster?

```
Already have image: e2e-runner
Fetching cluster endpoint and auth data.
kubeconfig entry generated for e2e-test-cluster.
Waiting consul port-forward to launch on 8500...
Forwarding from 127.0.0.1:8500 -> 8500
Handling connection for 8500
consul port-forward launched. Starting e2e tests...
Handling connection for 8500
Setting up lock at path: LockE2E/.lock
Attempting lock acquisition
Handling connection for 8500
Shutdown triggered or timeout during lock acquisition
```

**Comment by markmandel** [comment]
We've investigated, and these changes make the helm chart fail. It's not the e2e test.
(Can confirm locally as well). None of the CRDs install.

File: install/helm/agones/templates/crds/fleet.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (markmandel) [inline install/helm/agones/templates/crds/fleet.yaml]:
```suggestion
    "helm.sh/hook": "crd-install"
```

File: install/helm/agones/templates/crds/fleetallocation.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (markmandel) [inline install/helm/agones/templates/crds/fleetallocation.yaml]:
```suggestion
    "helm.sh/hook": "crd-install"
```

File: install/helm/agones/templates/crds/fleetautoscaler.yaml [hunk 3]
```diff
@@ -22,6 +22,8 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (markmandel) [inline install/helm/agones/templates/crds/fleetautoscaler.yaml]:
```suggestion
    "helm.sh/hook": "crd-install"
```

File: install/helm/agones/templates/crds/gameserver.yaml [hunk 4]
```diff
@@ -22,6 +22,8 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (markmandel) [inline install/helm/agones/templates/crds/gameserver.yaml]:
```suggestion
    "helm.sh/hook": "crd-install"
```

File: install/helm/agones/templates/crds/gameserverset.yaml [hunk 5]
```diff
@@ -22,6 +22,8 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (markmandel) [inline install/helm/agones/templates/crds/gameserverset.yaml]:
```suggestion
    "helm.sh/hook": "crd-install"
```

File: install/yaml/install.yaml [hunk 6]
```diff
@@ -163,6 +163,8 @@ metadata:
     chart: agones-0.6.0-rc
     release: agones-manual
     heritage: Tiller
+  annotations:
```
Comment (markmandel) [inline install/yaml/install.yaml]: Then this needs to be regenerated.

File: install/helm/agones/templates/crds/fleet.yaml (hunk 1 above)
Comment (smoya) [reply install/helm/agones/templates/crds/fleet.yaml]: Suggestion applied.

File: install/helm/agones/templates/crds/fleetallocation.yaml (hunk 2 above)
Comment (smoya) [reply install/helm/agones/templates/crds/fleetallocation.yaml]: Suggestion applied.

File: install/helm/agones/templates/crds/fleetautoscaler.yaml (hunk 3 above)
Comment (smoya) [reply install/helm/agones/templates/crds/fleetautoscaler.yaml]: Suggestion applied.

File: install/helm/agones/templates/crds/gameserver.yaml (hunk 4 above)
Comment (smoya) [reply install/helm/agones/templates/crds/gameserver.yaml]: Suggestion applied.

File: install/helm/agones/templates/crds/gameserverset.yaml (hunk 5 above)
Comment (smoya) [reply install/helm/agones/templates/crds/gameserverset.yaml]: Suggestion applied.

File: install/yaml/install.yaml (hunk 6 above)
Comment (smoya) [reply install/yaml/install.yaml]: Done!

**Comment by markmandel** [comment]
This keeps failing. Happy to dig in some more, but wanted to check - it's this working on your end? Wondering if there is some difference in setup here that I'm missing.

**Comment by markmandel** [comment]
Running another test - see if it passes. Also, was wrong about the rebase. That's weird. Not sure what is up.

**Pull request merged**